extensions/laracasts/src/laracasts.tsx

```diff
@@ -0,0 +1,114 @@
+import { ActionPanel, List, showToast, Action, Toast } from "@raycast/api";
+import { useEffect, useMemo, useState } from "react";
+import algoliaSearch from "algoliasearch";
+import _ from "lodash";
+
+const APPID = "1Z405N45FC";
+const APIKEY = "6c44626a6a8c21778291dc05232905e6";
+const INDEX = "lessons";
+
+type LaracastsLessonsHit = {
+  type: string;
+  title: string;
+  body: string;
+  path: string;
+  length: number;
+  difficulty: string;
+  thumbnail: string;
+  taxonomy: string;
+  objectID: string;
+  _highlightResult: {
+    content:
+      | {
+          value: string;
+          matchlevel: string;
+          fullyHighlighted: boolean;
+          matchedWords: string[];
+        }
+      | undefined;
+    hierarchy: {
+      [key: string]: {
+        value: string;
+        matchLevel: string;
+        matchedWords: string[];
+      };
+    };
+  };
+};
+
+const convertPathToURL = (path: string) => {
+  return `https://laracasts.com${path}`;
+};
+
+const convertTime = (seconds: number) => {
+  const h = Math.floor(seconds / 3600);
+  const m = Math.floor((seconds % 3600) / 60);
+
+  const hDisplay = h > 0 ? `${h}h ` : "";
+  const mDisplay = m > 0 ? `${m}m` : "";
+  return hDisplay + mDisplay;
+};
+
+export default function SearchLaracastsLessons() {
+  const algoliaClient = useMemo(() => {
+    return algoliaSearch(APPID, APIKEY);
+  }, [APPID, APIKEY]);
+
+  const algoliaIndex = useMemo(() => {
+    return algoliaClient.initIndex(INDEX);
+  }, [algoliaClient, INDEX]);
+
+  const [searchResults, setSearchResults] = useState<any[] | undefined>();
+  const [isLoading, setIsLoading] = useState(false);
```

Comment:
```suggestion
  const [isLoading, setIsLoading] = useState(true);
```
We need this to remove [empty state flicker](https://developers.raycast.com/basics/prepare-an-extension-for-store#empty-states)